### DISCUSSION TITLE
implementation naming consistency with upstream

### DISCUSSION THREAD

**Discussion by xenoterracide** [Netflix/dgs-framework#212] [Ideas]
why did DGS decide to prefix everything with `Dgs` in its API? why did it name things `DataFetcher` and `Data` instead of `DataLoader` and `Resolver` respectively? It would seem like it would have been better to use the naming conventions that Apollo uses (and other implementations?). Also, I might consider this a suggestion for the future. Also why "DGS", make it kind of hard to google.

**Comment by paulbakker** [comment]
Because naming things is hard 🤷🏻‍♂️
These choices were made very early on (long before we open sourced), and can't be changed without breaking existing code, so although we _could_ have made better naming choices in retrospect, it's unlikely going to change.
